src/components/TabSelector/TabSelector.tsx

```diff
@@ -1,42 +1,33 @@
-import PropTypes from 'prop-types';
-import React, {useCallback, useMemo, useState} from 'react';
+import {MaterialTopTabNavigationHelpers} from '@react-navigation/material-top-tabs/lib/typescript/src/types';
+import {TabNavigationState} from '@react-navigation/native';
+import React, {FunctionComponent, useCallback, useEffect, useMemo, useState} from 'react';
+import type {Animated} from 'react-native';
 import {View} from 'react-native';
-import _ from 'underscore';
+import {SvgProps} from 'react-native-svg';
 import * as Expensicons from '@components/Icon/Expensicons';
+import {LocaleContextProps} from '@components/LocaleContextProvider';
 import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
+import {RootStackParamList} from '@libs/Navigation/types';
 import CONST from '@src/CONST';
 import TabSelectorItem from './TabSelectorItem';
 
-const propTypes = {
+type TabSelectorProps = {
     /* Navigation state provided by React Navigation */
-    // eslint-disable-next-line react/forbid-prop-types
-    state: PropTypes.object.isRequired,
+    state: TabNavigationState<RootStackParamList>;
 
     /* Navigation functions provided by React Navigation */
-    navigation: PropTypes.shape({
-        navigate: PropTypes.func.isRequired,
-        emit: PropTypes.func.isRequired,
-    }).isRequired,
+    navigation: MaterialTopTabNavigationHelpers;
 
     /* Callback fired when tab is pressed */
-    onTabPress: PropTypes.func,
+    onTabPress?: (name: string) => void;
 
     /* AnimatedValue for the position of the screen while swiping */
-    position: PropTypes.shape({
-        interpolate: PropTypes.func.isRequired,
-    }),
+    position: Animated.AnimatedInterpolation<number | string>;
 };
 
-const defaultProps = {
-    onTabPress: () => {},
-    position: {
-        interpolate: () => {},
-    },
-};
-
-const getIconAndTitle = (route, translate) => {
+function getIconAndTitle(route: string, translate: LocaleContextProps['translate']): {icon: FunctionComponent<SvgProps>; title: string} {
```

Comment:
Let's create a separate type for return type `{icon: FunctionComponent<SvgProps>; title: string}`